esphome/components/b_parasite/sensor.py

```diff
@@ -0,0 +1,68 @@
+import esphome.codegen as cg
+import esphome.config_validation as cv
+from esphome.components import sensor, esp32_ble_tracker
+from esphome.const import (
+    CONF_BATTERY_VOLTAGE,
+    CONF_HUMIDITY,
+    CONF_ID,
+    CONF_MOISTURE,
+    CONF_MAC_ADDRESS,
+    CONF_TEMPERATURE,
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_VOLTAGE,
+    ICON_EMPTY,
+    UNIT_CELSIUS,
+    UNIT_PERCENT,
+    UNIT_VOLT,
+)
+
+CODEOWNERS = ["@rbaron"]
+
+DEPENDENCIES = ["esp32_ble_tracker"]
+
+b_parasite_ns = cg.esphome_ns.namespace("b_parasite")
+BParasite = b_parasite_ns.class_(
+    "BParasite", esp32_ble_tracker.ESPBTDeviceListener, cg.Component
+)
+
+CONFIG_SCHEMA = (
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(BParasite),
+            cv.Required(CONF_MAC_ADDRESS): cv.mac_address,
+            cv.Optional(CONF_TEMPERATURE): sensor.sensor_schema(
+                UNIT_CELSIUS, ICON_EMPTY, 1, DEVICE_CLASS_TEMPERATURE
+            ),
+            cv.Optional(CONF_HUMIDITY): sensor.sensor_schema(
+                UNIT_PERCENT, ICON_EMPTY, 3, DEVICE_CLASS_HUMIDITY
```

Comment:
```suggestion
                UNIT_PERCENT, ICON_EMPTY, 1, DEVICE_CLASS_HUMIDITY
```

That parameter is for number of decimals of accuracy the measurement provides - I highly doubt you can get more than 1% absolute accuray and maybe 0.1% relative accuracy without advanced hardware.

Reply:
Good point. Here we are converting the values to the [0, 1.0] interval (instead of [0, 100.0]), that's why I used 3 decimal points.
I realize now that this may not play great with the `UNIT_PERCENT` (which I believe expects the range [0, 100.0]). Is there an alternative unit of measure for the [0, 1.0] range (I didn't find a suitable one in `const.py`)? Alternatively I can just multiply the value by 100.0 before publishing it.

Reply:
Yes, multiply by 100 - then that's also consistent with the other humidity sensors in esphome

Reply:
Thanks - done.